[WIP] feat: add option to export wallets

dated opened the pull request:
## Proposed changes
<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
-->

Adds the option to export ones wallets.

## Types of changes
<!--
What types of changes does your code introduce?
_Put an `x` in the boxes that apply and remove the rest of them: keep this PR as concise, but descriptive, as possible.._
-->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (improve a current implementation without adding a new feature or fixing a bug)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build (changes that affect the build system)
- [ ] Docs (documentation only changes)
- [ ] Test (adding missing tests or fixing existing tests)
- [ ] Other... Please describe:

## Checklist
<!--
_Put an `x` in the boxes that apply and remove this text and the rest of them. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._
-->

- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
<!--

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...
-->

dated commented:
any feedback is appreciated @alexbarnsley @j-a-m-l @ItsANameToo

dated commented:
I'll add a notice to the modal regarding the passphrase, for the import a 'Batch Import' button could be added to the import page?

j-a-m-l commented:
I'm not sure about that option; we could rename the buttons to "Import Wallets", but that page would need a modal to import the wallets and it would break the step style and behaviour of the process, so it wouldn't fit there smoothly.

I'd rather add other button. Maybe "Batch Import", near the "Import Wallet" button is clear enough.

zillionn commented:
I guess this will be a nice feature when you're switching computers or reinstalling the OS, right? If yes, how about to export the whole `Profile` with its settings and everything? Then when you install the desktop wallet on the new comp, you'll have an option to import the profile.

j-a-m-l commented:
Yes, exporting the whole profile could be a feature. In that case it would include data to import the network in case it doesn't exist yet.

dated commented:
I added the following warning to the modal:

>Your exported wallets will not contain your passphrases, only the addresses and respective names will be saved!

j-a-m-l commented:
It's ignoring Ledger wallets, even when they have a custom name.

dated commented:
Does this work for you @j-a-m-l? I added an additional option for the ledger wallets.

j-a-m-l commented:
@dated it works OK, but the Ledger switch should be visible or enabled only when there is a Ledger connected.

Apart from that, I'd change the format to use an object as the value instead of the custom name of the wallet, so the format is more flexible:
```
  "address": {
    "name": "my wallet"
  }
```

Related to that, I'd export other data per wallet that could be useful for other purposes (trading, taxes, etc.), such as:
 - public key
 - delegate username
 - voted delegate
 - current balance
 - current balance in the configured currency (BTC, EUR, etc.).
 - number of transactions

@alexbarnsley @luciorubeens @boldninja what do you think?

zillionn commented:
IMO there should be an option to select/deselect every object before exporting. Maybe something like this, with checkboxes:

![1](https://user-images.githubusercontent.com/3462869/58123359-517f5400-7c14-11e9-89f9-7682a1490fdb.png)

dated commented:
The ledger option is now disabled if no ledger is connected and I've added some more fields for now:

```
  "wallets": [
    {
      "ANoNeeFDFo61XcQBYad6rEbPvfATwgjTNT": {
        "name": "test",
        "publicKey": "02ead78d2ecd00a966f45c1ea3fb8d2e032dcb704a451a1af0426789261d37459a",
        "balance": {
          "ARK": "1780.56728118",
          "USD": "1105.27"
        }
      }
    }
  ]
```

I'll try to add the remaining fields and some tests tomorrow.

dated commented:
I would refrain from the transaction counts for now as the API does return wrong values, and I don't see much use for that information anyway.

The other fields have been added. I removed the address as key and added it as a property. Are you ok with the current format @j-a-m-l?

luciorubeens commented on src/renderer/components/Modal/ModalExportWallets.vue:
```diff
@@ -0,0 +1,265 @@
+        },
+        excludeLedger: {
+          active: false,
+          isDisabled: !this.isLedgerConnected,
```
The `data` can not access the `computed` properties, you need to use the `watch`:

```js
watch: {
  isLedgerConnected: {
    handler(val) {
      this.options.excludeLedger.isDisabled = !val
    },
    immediate: true
  }
}
```

dated commented:
Thanks for the review @luciorubeens, I'll get the changes done asap.

dated commented:
All fixed, thanks again @luciorubeens.
